edocumentation-domain/src/main/java/org/dcsa/edocumentation/domain/persistence/state/BookingStateMachineTables.java

```diff
@@ -0,0 +1,48 @@
+package org.dcsa.edocumentation.domain.persistence.state;
+
+import lombok.experimental.UtilityClass;
+import org.dcsa.edocumentation.domain.persistence.entity.enums.BkgDocumentStatus;
+
+import java.util.Map;
+import java.util.Set;
+
+import static org.dcsa.edocumentation.domain.persistence.entity.enums.BkgDocumentStatus.*;
+
+@UtilityClass
+class BookingStateMachineTables {
+
+  static final Map<BkgDocumentStatus, Set<BkgDocumentStatus>> BOOKING_STATE_TRANSITION_TABLE = buildStateTransitionTable();
+  static final BkgDocumentStatus INITIAL_STATE = RECE;
+
+  private static Set<BkgDocumentStatus> successorStates(BkgDocumentStatus ... statuses) {
+    Set<BkgDocumentStatus> successors = Set.of(statuses);
+    if (successors.isEmpty()) {
+      // Fail-safe
+      throw new IllegalArgumentException("No successor states!? Please us terminalState instead");
```

Comment:
should we throw DCSA exceptions here

Reply:
It is intended as a bootstrap exception, but I suppose it can be triggered on the first request if the class is lazily loaded.  Will change these as well.